### PR TITLE
Fix a bug in Conv+Activation fusion

### PR DESCRIPTION
**Pull request by duli2012**

### CODE REVIEW

**Comment by snnn** [comment]
Today I found a bug. It looks like some nodes was fused and the new graph is not valid.  It is only observed when cuda is enabled. And it's a new bug just introduced in this week. I haven't found the root cause.

File: onnxruntime/core/graph/conv_activation_fusion.cc [hunk 1]
```diff
@@ -14,6 +14,33 @@ namespace {
...
+  //remove output edge of activation
+  for (auto& output_edge : output_edges) {
+    NodeIndex dst_node_index = output_edge.GetNode().Index();
+    int src_arg_index = output_edge.GetSrcArgIndex();
+    int dst_arg_index = output_edge.GetDstArgIndex();
+    g.RemoveEdge(act.Index(), dst_node_index, src_arg_index, dst_arg_index);
+  }
+
+  //connect fused_conv node and nodes after activation nodes
+  for (auto& out : output_edges) {
```
Comment (weixingzhang) [inline onnxruntime/core/graph/conv_activation_fusion.cc]: These two loops body can be merged.

File: onnxruntime/core/graph/conv_activation_fusion.cc [hunk 2]
```diff
@@ -14,6 +14,33 @@ namespace {
...
+
+void HandleActNodeEdges(Graph& g, const Node& act, Node& fused_conv) {
+  Node::EdgeSet output_edges, input_edges;
+  for (auto it = act.OutputEdgesBegin(); it != act.OutputEdgesEnd(); ++it) {
+    output_edges.insert(*it);
+  }
+
+  for (auto it = act.InputEdgesBegin(); it != act.InputEdgesEnd(); ++it) {
+    input_edges.insert(*it);
+  }
```
Comment (weixingzhang) [inline onnxruntime/core/graph/conv_activation_fusion.cc]: It seems that you don't need to have a local copy for input edges.

File: onnxruntime/core/graph/conv_activation_fusion.cc [hunk 3]
```diff
@@ -14,6 +14,33 @@ namespace {
 bool IsFusableActivation(const Node& node) {
   return utils::IsSupportedOptypeVersionAndDomain(node, "LeakyRelu", 6) || utils::IsSupportedOptypeVersionAndDomain(node, "Relu", 6) || utils::IsSupportedOptypeVersionAndDomain(node, "Sigmoid", 6) || utils::IsSupportedOptypeVersionAndDomain(node, "Tanh", 6);
 }
+
+void HandleActNodeEdges(Graph& g, const Node& act, Node& fused_conv) {
```
Comment (weixingzhang) [inline onnxruntime/core/graph/conv_activation_fusion.cc]: I would suggest to rename it to HandleActivationNodeEdges.

**Comment by duli2012** [comment]
@snnn this fusion is not enabled by default, so it's unlikely to fail the build.

**Pull request merged**